### ISSUE TITLE
Coredump when doing a user switch

### ISSUE THREAD

**Issue by ulrichSchreiner** [f-koehler/KTailctl#182]
hi,

when i do a [fast user switch](https://tailscale.com/kb/1225/fast-user-switching) (yes i know this is currently alpha) `ktailctl` core dumps.

i enter
~~~
$ sudo tailscale switch <nickname>
~~~

and get this in the journal
~~~
Apr 01 17:24:35 myhost krunner[267483]: Logging to: /home/usc/.var/app/org.fkoehler.KTailctl/.local/state/org.fkoehler.KTailctl.log
Apr 01 17:24:35 myhost krunner[267483]: [2024-04-01T17:24:35][warning] QCoreApplication::postEvent: Unexpected null receiver
Apr 01 17:24:35 myhost krunner[267483]: Logging to: /home/usc/.var/app/org.fkoehler.KTailctl/.local/state/org.fkoehler.KTailctl.log
Apr 01 17:24:35 myhost krunner[267483]: [2024-04-01T17:24:35][warning] QCoreApplication::postEvent: Unexpected null receiver
Apr 01 17:24:35 myhost sudo[269636]: pam_unix(sudo:session): session closed for user root
Apr 01 17:24:36 myhost (sd-parse-elf)[269792]: Could not parse number of program headers from core file: invalid `Elf' handle
Apr 01 17:24:36 myhost systemd-coredump[269690]: [🡕] Process 267483 (ktailctl) of user 1000 dumped core.

                                                        Module /app/bin/ktailctl without build-id.
                                                        Module /app/bin/ktailctl
                                                        Module /app/lib/qml/org/kde/kirigamiaddons/delegates/libdelegatesplugin.so without build-id.
                                                        Module /app/lib/libtailwrap_logging_wrapper.so without build-id.
                                                        Stack trace of thread 2:
                                                        #0  0x00007aa0d20e1b08 n/a (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5.15.10 + 0x2e1b08)
                                                        #1  0x0000000000000000 n/a (n/a + 0x0)
                                                        ELF object binary architecture: AMD x86-64
~~~

The logfile `KTailctl.log` contains these two lines

~~~
[2024-04-01T17:24:35][warning] QCoreApplication::postEvent: Unexpected null receiver
[2024-04-01T17:24:35][warning] QCoreApplication::postEvent: Unexpected null receiver
~~~

OS: Arch Linux, Kernel 6.8.2
~~~
❯ flatpak info org.fkoehler.KTailctl

Fabian Köhler - GUI for Tailscale mesh network

         Kennung: org.fkoehler.KTailctl
             Ref: app/org.fkoehler.KTailctl/x86_64/stable
     Architektur: x86_64
           Zweig: stable
         Version: 0.15.0
          Lizenz: GPL-3.0+
        Ursprung: flathub
        Sammlung: org.flathub.Stable
    Installation: system
     Installiert: 16,1 MB
Laufzeitumgebung: org.kde.Platform/x86_64/5.15-23.08
             Sdk: org.kde.Sdk/x86_64/5.15-23.08

          Commit: 2731e948c73b69e480fc9f0ba024c22b63019b81f479b2afd71019711d18e517
          Parent: 5b027e2efbfec2accb09ef42d5170f423beaa811682716ad9073a32dddd9157d
         Betreff: Release v0.15.0. (d5806366)
           Datum: 2024-02-18 06:32:29 +0000
~~~

**Comment by f-koehler** [comment]
Thanks for the early report, I was not aware of this new feature. I will see if I can create a second Tailscale account and reproduce the issue. Also I want to implement some multi-account functionality into KTailctl at some point as well.

**Comment by f-koehler** [comment]
I just release `v0.16.0` of KTailctl which should be available on flathub in a few hours. This update - among other changes - contains the current version of the Tailscale go module. I have not tested the fast user switching and do not know in which version they introduced it, but maybe just this change fixes the crash.

I would really appreciate if you could try it and report whether it still crashes. But no pressure I will still try reproduce this issue when I have some free time.

If you are very eager you can try the new version before it is available on flathub:
```
flatpak install --user https://dl.flathub.org/build-repo/94283/org.fkoehler.KTailctl.flatpakref
```

**Comment by ulrichSchreiner** [comment]
hi,

sorry for the delay, i was on the way the whole day.

i installed the new version on flathub and it works in one direction, not the other :smile: 

i can switch from my work-tailnet to my home-tailnet. the icon of ktailctl short flashes and then everything works. i can also open the window of `ktailctl` and the old peers disappear and the new ones appear. great.

but when i switch back (to my work-tailnet) again it still core dumps. but now it looks a little different:

~~~
 systemd-coredump[351112]: [🡕] Process 350089 (ktailctl) of user 1000 dumped core.

    Module /app/bin/ktailctl without build-id.
    Module /app/bin/ktailctl
    Module /app/lib/qml/org/kde/kirigamiaddons/components/libcomponentsplugin.so without build-id.
    Module /app/lib/qml/org/kde/kirigamiaddons/components/libcomponentsplugin.so
    Module /app/lib/qml/org/kde/kirigamiaddons/labs/components/libcomponentslabsplugin.so without build-id.
    Module /app/lib/qml/org/kde/kirigamiaddons/labs/components/libcomponentslabsplugin.so
    Module /app/lib/qml/org/kde/kirigamiaddons/formcard/libformcardplugin.so without build-id.
    Module /app/lib/qml/org/kde/kirigamiaddons/formcard/libformcardplugin.so
    Module /app/lib/qml/org/fkoehler/KTailctl/Components/libmycomponentsplugin.so without build-id.
    Module /app/lib/qml/org/fkoehler/KTailctl/Components/libmycomponentsplugin.so
    Module /app/lib/qml/org/kde/kirigamiaddons/delegates/libdelegatesplugin.so without build-id.
    Module /app/lib/qml/org/kde/kirigamiaddons/delegates/libdelegatesplugin.so
    Module /app/lib/libtailwrap_logging_wrapper.so without build-id.
    Stack trace of thread 2:
    #0  0x00007719ead81335 n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x181335)
    #1  0x00007719ead81470 n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x181470)
    #2  0x00005fa221087f28 n/a (/app/bin/ktailctl + 0x34f28)
    #3  0x00005fa221088376 n/a (/app/bin/ktailctl + 0x35376)
    #4  0x00005fa2210696e0 n/a (/app/bin/ktailctl + 0x166e0)
    #5  0x00007719ec1c4e2a n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x3c4e2a)
    #6  0x00007719ec052f2b n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x252f2b)
    #7  0x00007719ec0558a5 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x2558a5)
    #8  0x00007719ec07102a n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x27102a)
    #9  0x000077199035deb4 n/a (n/a + 0x0)
    #10 0x00007719ec0b26c4 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x2b26c4)
    #11 0x00007719ebffe3c6 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x1fe3c6)
    #12 0x00007719ebffe9cb n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x1fe9cb)
    #13 0x00007719ec187327 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x387327)
    #14 0x00007719ec105643 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x305643)
    #15 0x00007719ec105ff0 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x305ff0)
    #16 0x00007719ec1b314c n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x3b314c)
    #17 0x00007719eaddf530 n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x1df530)
    #18 0x00007719ec2b6843 n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x4b6843)
    #19 0x00007719ec2b6afc n/a (/usr/lib/x86_64-linux-gnu/libQt6Qml.so.6.6.2 + 0x4b6afc)
    #20 0x00007719edb9be98 n/a (/usr/lib/x86_64-linux-gnu/libQt6Widgets.so.6.6.2 + 0x19be98)
    #21 0x00007719ead7ec68 n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x17ec68)
    #22 0x00007719ead82c15 n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x182c15)
    #23 0x00007719eb060a0f n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x460a0f)
    #24 0x00007719e9f60e59 n/a (/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7600.6 + 0x60e59)
    #25 0x00007719e9fbe2b8 n/a (/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7600.6 + 0xbe2b8)
    #26 0x00007719e9f5e513 n/a (/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.7600.6 + 0x5e513)
    #27 0x00007719eb0602bf n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x4602bf)
    #28 0x00007719ead8babb n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x18babb)
    #29 0x00007719ead87d1d n/a (/usr/lib/x86_64-linux-gnu/libQt6Core.so.6.6.2 + 0x187d1d)
    #30 0x00005fa2210791c5 n/a (/app/bin/ktailctl + 0x261c5)
    #31 0x00007719ea63b08a n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x2808a)
    #32 0x00007719ea63b14b n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x2814b)
    #33 0x00005fa221065845 n/a (/app/bin/ktailctl + 0x12845)
    ELF object binary architecture: AMD x86-64
~~~

at first i thought the problem was, that my account was not  the operator. but i changed this, now i am the operator in both tailnets.

when i switch to my home-net there are log-entries in your logfile:
~~~
[2024-04-02T18:18:43][critical][/run/build/ktailctl/src/statistics.cpp:48] No down speed statistics for peer ""
[2024-04-02T18:18:43][critical][/run/build/ktailctl/src/statistics.cpp:39] No up speed statistics for peer ""
[2024-04-02T18:18:43][warning][qrc:Peers.qml:91] qrc:Peers.qml:91: TypeError: Cannot read property 'average1Second' of null
[2024-04-02T18:18:43][warning][qrc:Peers.qml:99] qrc:Peers.qml:99: TypeError: Cannot read property 'average1Second' of null
~~~
but: there are `critical` log entries but the switch works. but when switching to my work-net, `ktailctl` crashes and NO logs are written. 

in both tailnets i can start/stop `ktailctl` without any problems, but the switch works only when i am in a specific tailnet.


Version:
~~~
❯ flatpak info org.fkoehler.KTailctl

Fabian Köhler - GUI for Tailscale mesh network

         Kennung: org.fkoehler.KTailctl
             Ref: app/org.fkoehler.KTailctl/x86_64/stable
     Architektur: x86_64
           Zweig: stable
         Version: 0.16.0
          Lizenz: GPL-3.0+
        Ursprung: flathub
        Sammlung: org.flathub.Stable
    Installation: system
     Installiert: 17,9 MB
Laufzeitumgebung: org.kde.Platform/x86_64/6.6
             Sdk: org.kde.Sdk/x86_64/6.6

          Commit: f3e892bea32e8c80d462a5422437a1e579445d851f909e1c072a5d79e4454e48
          Parent: 2731e948c73b69e480fc9f0ba024c22b63019b81f479b2afd71019711d18e517
         Betreff: Fix commit hash for KTailctl. (e8fb2811)
           Datum: 2024-04-01 17:10:48 +0000
~~~

ATM there is an outstanding update from qt6 (i have 6.6.2 but 6.6.3 is pending), but i doubt this will make a great difference.

please dont invest too much time in this issue. fast user switching was introduced in decemter 2022 (https://tailscale.com/blog/fast-user-switching) but it is still alpha so it can change in the future.

**Comment by ulrichSchreiner** [comment]
the new 0.17.1 (perhaps 0.17 too) fixed the issue. now switching the user works.

**Comment by f-koehler** [comment]
> the new 0.17.1 (perhaps 0.17 too) fixed the issue. now switching the user works.

Great news! That is not surprising as I rewrote the whole back-end for 0.17 to be much simpler and robust. I did not test as I still need to set up some secondary account to play with.

I am planning to expose the account switching in the interface at some point in the future but first want to tackle taildrop and network statistics. For now it is good enough that it does not crash :)
